core/lb.c

```diff
@@ -377,14 +386,37 @@ static struct oio_lb_pool_vtable_s vtable_LOCAL =
 static struct _lb_item_s *
 _item_make (oio_location_t location, const char *id, const char *addr)
 {
-	const size_t len = strlen (id);
-	struct _lb_item_s *out = g_malloc0 (sizeof(struct _lb_item_s) + len + 1);
+	struct _lb_item_s *out = g_malloc0(sizeof(struct _lb_item_s));
 	out->location = location;
-	g_strlcpy (out->addr, addr, sizeof(out->addr));
-	memcpy(out->id, id, len + 1);
+	g_strlcpy(out->addr, addr, sizeof(out->addr));
+	g_strlcpy(out->id, id, sizeof(out->id));
 	return out;
 }
 
+static struct oio_lb_selected_item_s *
+_item_select(const struct _lb_item_s *src)
+{
+	struct oio_lb_selected_item_s *res =  g_malloc0(sizeof *res);
+	if (src != NULL) {
+		res->item = g_malloc0(sizeof(struct oio_lb_item_s));
+		g_strlcpy(res->item->addr, src->addr, sizeof(res->item->addr));
+		g_strlcpy(res->item->id, src->id, sizeof(res->item->id));
+		res->item->location = src->location;
+		res->item->weight = src->weight;
+	}
+	return res;
+}
+
+static void
+_selected_item_free(struct oio_lb_selected_item_s *src)
+{
+	g_free((gpointer)src->item);
+	g_free((gpointer)src->expected_slot);
+	g_free((gpointer)src->final_slot);
+	memset(src, 0, sizeof(*src));
```

Comment:
is it really necessary to memset ?

Reply:
It's not necessary, it's just a way to avoid bugs.